models/masters.model.js

```diff
@@ -36,15 +40,192 @@ export default {
     })
     return master
   },
-  addMaster: async (name, rating) => {
-    const newMaster = await Master.create({ name, rating })
-    return newMaster
+  getFreeMasters: async (startTime, endTime, cityId) => {
+    const busyMasters = await Order.findAll({
+      attributes: ['masterId'],
+      where: {
+        [Op.or]: {
+          [Op.and]: {
+            startTime: {
+              [Op.lte]: startTime
+            },
+            endTime: {
+              [Op.gte]: endTime
+            }
+          },
+          endTime: {
+            [Op.and]: {
+              [Op.gt]: startTime,
+              [Op.lt]: endTime
+            }
+          },
+          startTime: {
+            [Op.and]: {
+              [Op.gt]: startTime,
+              [Op.lt]: endTime
+            }
+          }
+        }
+      }
+    })
+    const AllMasters = await Master.findAll({
+      include: [
+        {
+          where: { id: cityId },
+          model: City,
+          as: 'cities'
+        }
+      ]
+    })
+    const busyMastersId = busyMasters.map((master) => master.dataValues.masterId)
+    const freeMasters = AllMasters.filter((master) => !busyMastersId.includes(master.dataValues.id))
```

Comment:
Use not in
`[Op.notIn]: [1, 2]`
https://sequelize.org/docs/v6/core-concepts/model-querying-basics/#operators